### DISCUSSION TITLE
why "lineGap * 3" ?

### DISCUSSION THREAD

**Discussion by sdragonx** [nothings/stb#1359] [Q&A]
stbtt:
float height()const
{
    int lineGap;
    stbtt_GetFontVMetrics(handle(), nullptr, nullptr, &lineGap);
    return (**lineGap** * 3) * m_scale;
}

freetype2:
float height()const
{
    const FT_Size_Metrics& mat = m_face->size->metrics;
    return mat.height * INV_64;
}

The results of these two functions are the same.

**Comment by sdragonx** [comment]
and how stbtt.dpi == f(freetype2.dpi) ?

freetype2:
set_size(float size, int dpi = 72)
FT_Set_Char_Size(m_face, 0, FT_F26Dot6(size) << 6, dpi, dpi);

stbtt:
set_size(float size, int dpi = 72)
float n = size * dpi / 72.0f;
//float n = size * dpi / 71.775f; //err
float scale = stbtt_ScaleForMappingEmToPixels(handle(), n);

there is some error in the value.

**Reply by nothings** [reply]
The code was written 13 years ago. If there's no comment explaining why something is the way it is, I can't tell you.

What I can tell you was that it was written WITHOUT looking at FreeType, for obvious copyright reasons. So it was written based on the specification and, because the specification is incomplete about some of these issues, based on experimenting to attempt to get the right results.
